docarray/typing/url/audio_url.py

```diff
@@ -0,0 +1,114 @@
+import wave
+from typing import TYPE_CHECKING, Any, Type, TypeVar, Union
+
+import numpy as np
+
+from docarray.typing.url.any_url import AnyUrl
+
+if TYPE_CHECKING:
+    from pydantic import BaseConfig
+    from pydantic.fields import ModelField
+
+    from docarray.proto import NodeProto
+
+T = TypeVar('T', bound='AudioUrl')
+
+AUDIO_FILE_FORMATS = ['wav']
+MAX_INT_16 = 2**15
+
+
+class AudioUrl(AnyUrl):
+    """
+    URL to a .wav file.
+    Can be remote (web) URL, or a local file path.
+    """
+
+    def _to_node_protobuf(self: T) -> 'NodeProto':
+        """Convert Document into a NodeProto protobuf message. This function should
+        be called when the Document is nested into another Document that needs to
+        be converted into a protobuf
+
+        :return: the nested item protobuf message
+        """
+        from docarray.proto import NodeProto
+
+        return NodeProto(audio_url=str(self))
+
+    @classmethod
+    def validate(
+        cls: Type[T],
+        value: Union[T, np.ndarray, Any],
+        field: 'ModelField',
+        config: 'BaseConfig',
+    ) -> T:
+        url = super().validate(value, field, config)  # basic url validation
+        has_audio_extension = any(ext in url for ext in AUDIO_FILE_FORMATS)
+        if not has_audio_extension:
+            raise ValueError(
+                f'Audio URL must have one of the following extensions:'
+                f'{AUDIO_FILE_FORMATS}'
+            )
+        return cls(str(url), scheme=None)
+
+    def load(self: T) -> np.ndarray:
```

Comment:
I would say `AudioNdArray`.
Normally we just load the np.ndarray, bc the framework specific conversion can happen when putting it back into the document, and having it as np.ndarray makes the least assumptions about the sorrounding code.
In this case, since there are actual audio features that come with the array, I would go with `AudioNdArray`. It can still be treated like a normal np.ndarray, but bring the aforementioned features.

But just verify in a test that setting a `AudioNdArray` to a field with type `AudioTorchArray` actually works without issue?

Reply:
great I like it better like this